MaraBot/Commands/LeaderboardCommandModule.cs

```diff
@@ -50,8 +60,33 @@ public async Task Execute(CommandContext ctx, int weekNumber = -1)
                 return;
             }
 
-            await Display.LeaderboardAsync(ctx, weekly, weekNumber == currentWeek);
-            await CommandUtils.SendSuccessReaction(ctx);
+            if (weekNumber == currentWeek)
+            {
+                var spoilerChannel = ctx.Guild.Channels
+                    .FirstOrDefault(channel => Config.WeeklySpoilerChannel.Equals(channel.Value.Name));
+
+                if (spoilerChannel.Value == null)
+                {
+                    await ctx.RespondAsync(
+                        "No spoiler channel set.\n" +
+                        "This shouldn't happen! Please contact your friendly neighbourhood developers!");
+                }
+                else if (ctx.Channel.Equals(spoilerChannel.Value))
+                {
+                    await ctx.RespondAsync(Display.LeaderboardEmbed(ctx, weekly, false));
+                    await CommandUtils.SendSuccessReaction(ctx);
+                }
+                else
+                {
+                    await ctx.RespondAsync("This week's leaderboard can only be displayed on the spoiler channel!");
+                    await CommandUtils.SendFailReaction(ctx);
+                }
+            }
+            else
+            {
+                await ctx.RespondAsync(Display.LeaderboardEmbed(ctx, weekly, false));
+                await CommandUtils.SendSuccessReaction(ctx);
+            }
```

Comment:
Given the above suggestion is implemented, I prefer this code.
If you like having `else` more, you can keep it. :)
Also, fixed some code duplication by changing around the order.
```suggestion
                if (!ctx.Channel.Equals(spoilerChannel.Value))
                {
                    await ctx.RespondAsync("This week's leaderboard can only be displayed on the spoiler channel!");
                    await CommandUtils.SendFailReaction(ctx);
                    return;
                }
            }

            await ctx.RespondAsync(Display.LeaderboardEmbed(ctx, weekly, false));
            await CommandUtils.SendSuccessReaction(ctx);
```